subsys/spm/Kconfig

```diff
@@ -71,6 +71,20 @@ config SPM_SERVICE_READ
 
 endif
 
+config SPM_ALLOW_NON_SECURE_RESET
+	bool "Allow Non-Secure reset"
+#FIXME: Change this to Default to `n` when the issue regarding doing a software
+#	reset from non-secure state with a secure debugger is resolved.
+	default y if BOARD_NRF9160_PCA20035
```

Comment:
I would suggest that both secure and non-secure are able to trigger soft-reset. Any users that require a soft-reset to go through secure region shall know what they are doing, and should then add this config entry manually.

My suggestion is to set this default y in all cases, allowing both regions to execute soft-reset.

Reply:
@ioannisg any objections?

Reply:
This sort of contradicts the long-term strategy of least-privilege policy. For example, we allocate to non-secure only the resources we are using, right? Same fore reset; this is, really, a secure side thing: Non-Secure should not be able to reboot the SoC, if you we don't know what the Secure side is doing. So it's OK to have this as configurable option (not ideal), but not default-y.

Reply:
Right now we need to look at usability vs security. It doesn't help how secure we makes our examples (by default configuration) if it leads to our customer having problem running/ debugging it. We should have a solution that can be used by our customer without having to use DevZone /Support.
My take on this is make user friendly bu default and add documentation describing what you should have when going into production.